### PR TITLE
o/snapstate: enable split refresh for auto-refresh

### PR DESCRIPTION
**Pull request by MiguelPires**
Enable splitting the single reboot refresh for hybrid systems in the auto-refresh scenario.

### CODE REVIEW

File: overlord/snapstate/snapstate.go [hunk 1]
```diff
@@ -3242,7 +3241,20 @@ func autoRefreshPhase2(ctx context.Context, st *state.State, updates []*refreshC
...
+	var updateTss *UpdateTaskSets
+	var updated []string
+	if essential, nonEssential, ok := canSplitRefresh(deviceCtx, toUpdate); ok {
+		// if we're on classic with a kernel/gadget, split installs with essential
+		// snaps and apps so that the apps don't have to wait for a reboot
+		updateFunc := func(updates []minimalInstallInfo) ([]string, *UpdateTaskSets, error) {
+			// extra names are ignored so it's fine to passed all of them in each call
+			return doUpdate(ctx, st, nil, toUpdate, nil, userID, flags, nil, deviceCtx, fromChange)
+		}
+		updated, updateTss, err = splitRefresh(st, essential, nonEssential, userID, flags, updateFunc)
```
Comment (pedronis) [inline overlord/snapstate/snapstate.go:3253]: afaict this code is not hit by tests

**Review by pedronis — COMMENTED** [review]
new added code seems untested

File: overlord/snapstate/snapstate.go (hunk 1 above)
Comment (MiguelPires) [reply overlord/snapstate/snapstate.go]: I've correct the test now, thank you

**Review by pedronis — APPROVED** [review]
thanks

**Review by andrewphelpsj — APPROVED** [review]
thanks!

**Pull request merged**